src/artists.ts

```diff
@@ -0,0 +1,61 @@
+import * as vscode from 'vscode';
+import SpotifyWebApi = require('spotify-web-api-node');
+
+export class ArtistsProvider implements vscode.TreeDataProvider<ArtistsAndTracks> {
+
+  constructor(private spotify: SpotifyWebApi) {}
+
+  async getChildren(element?: ArtistsAndTracks): Promise<ArtistsAndTracks[] | undefined> {
+    if (element === undefined) {
+      return this.getArtistsAndTracks(this.spotify);
+    }
+    return element.children;
+  }
+
+  async getTreeItem(element: ArtistsAndTracks): Promise<ArtistsAndTracks> {
+    return element;
+  }
+
+  async getArtistsAndTracks(spotify: SpotifyWebApi) {
+    try {
+      const artists = await this.spotify.getFollowedArtists();
+      return artists.body.artists.items.map((data) => new ArtistsAndTracks(data, this.spotify));
+      } catch (err: any) {
+      vscode.window.showErrorMessage(err);
+    }
+  }
+}
+
+
+export class ArtistsAndTracks extends vscode.TreeItem {
+  name: string;
+  uri: string;
+  children: Promise<ArtistsAndTracks[]> | undefined;
+
+  constructor (public readonly data: any, spotify?: SpotifyWebApi, parent?: ArtistsAndTracks) {
+    super(data);
```

Comment:
I'm assuming this is the proper label given the code below
```suggestion
    super(data.name);
```